api/src/test/java/org/apache/iceberg/types/TestReadabilityChecks.java

```diff
@@ -70,7 +70,7 @@ public void testPrimitiveTypes() {
             CheckCompatibility.writeCompatibilityErrors(
                 new Schema(required(1, "to_field", to)), fromSchema);
 
-        if (TypeUtil.isPromotionAllowed(from, to)) {
+        if (TypeUtil.isPromotionAllowed(from, to, 2)) {
```

Comment:
This means date promotions will fail in this test (as expected). But should there be a v3-specific test showing date promotions work?